py-rse/git-cmdline.Rmd

```diff
@@ -195,30 +232,78 @@ No commits yet
 Changes to be committed:
   (use "git rm --cached <file>..." to unstage)
 
-	new file:   names.txt
-```
-
-Git now knows that it's supposed to keep track of `names.txt`,
-but it hasn't actually recorded any changes yet.
-To do taht,
-we need to tell it to [commit][commit] our work:
-
+        new file:   bin/book_summary.sh
+        new file:   bin/collate.py
+        new file:   bin/countwords.py
+        new file:   bin/mymodule.py
+        new file:   bin/plotcounts.py
+        new file:   data/anne_of_green_gables.txt
+        new file:   data/common_sense.txt
+        new file:   data/count_of_monte_cristo.txt
+        new file:   data/dracula.txt
+        new file:   data/emma.txt
+        new file:   data/ethan_frome.txt
+        new file:   data/frankenstein.txt
+        new file:   data/jane_eyre.txt
+        new file:   data/life_of_frederick_douglass.txt
+        new file:   data/moby_dick.txt
+        new file:   data/mysterious_affair_at_styles.txt
+        new file:   data/pride_and_prejudice.txt
+        new file:   data/sense_and_sensibility.txt
+        new file:   data/sherlock_holmes.txt
+        new file:   data/time_machine.txt
+        new file:   data/treasure_island.txt
+        new file:   results/anne_of_green_gables.csv
+        new file:   results/common_sense.csv
+```
+
+Adding all your existing files is a dangerous operation,
```

Comment:
So it looks like we're making the decision to track/version not just our scripts in the `bin` directory, but also all our data files and all our result files.

Do we need to add some discussion here about that fact that while you always want to version your scripts, it's not always practical or desirable to version data files and result files (i.e. data files and result files can be very large and/or a binary format)? 

(The final challenge in the #325 scripting lesson requires people to create `~/zipfs-law/results/emma.png`, so this will also need to be addressed here - I'm guessing we wouldn't version .png files?)

Reply:
I'm adding a paragraph that points out that we've chosen to version everything, but that isn't always the right decision, and a pointer to the chapter on Project Structure which talks more about why things should or shouldn't be in version control.